pkg/plugins/repo/service.go

```diff
@@ -102,14 +102,14 @@ func (m *Manager) grafanaCompatiblePluginVersions(pluginID string, compatOpts Co
 		return nil, err
 	}
 
-	u.Path = path.Join(u.Path, "repo", pluginID)
+	u.Path = path.Join(u.Path, pluginID, "versions")
```

Comment:
I'm not sure if this endpoint supports grafana depenency checks like I think the repo one does given Grafana provides HTTP request headers with Grafana version and architecture. @wbrowne do you remember?

Reply:
Yep it does. The Repo endpoint is key for getting version information so that gcom do some checks (based on arch,os and grafana version) and also gives us enough version information for us to check too.

Reply:
> The Repo endpoint is key for getting version information so that gcom do some checks (based on arch,os and grafana version) and also gives us enough version information for us to check too.

The checks (arch, os, grafana version) were also done here on the client side so we don't need to rely in gcom to do so.

In any case, I noticed that the /versions endpoint doesn't return arch information. It has a `packages` section but it's always empty, which seems like a bug (e.g. https://grafana.com/api/plugins/grafana-athena-datasource/versions).

So since we need to modify GCOM anyway, we can either allow internal plugins for the `repo` endpoint or we fix the `packages` information for the `versions` endpoint. I am more willing to do the second because I don't know if something else uses the `repo` endpoint and if that would be a breaking change.

Reply:
Updating `versions` sounds like a safer move alright. It would be good to compare the two endpoints behaviour then as IIRC repo was intended to be the new API to use (which I'm guessing replaced `versions`?)

Reply:
> It would be good to compare the two endpoints behaviour

As far as I can see, the differences are that `/repo`:

 - Has more information, apart from versions (or with a different format) but both have all the info we need here.
 - Does that validation (arch, version, etc) based on headers (not sure why), which we also do in the client
 - Filters out `internal` plugins

I'm curious if you say that `repo` is more recent / expected to be used. Can I look for info about that somewhere?

Reply:
Apologies I missed this reply. IIRC it was @DanCech who explained this back when we were figuring out what was the most appropriate API when working on the runtime plugin installation stuff. Please correct my understanding if I'm wrong Dan!

Reply:
Opened a PR for this https://github.com/grafana/grafana-com/pull/8029

> * Does that validation (arch, version, etc) based on headers (not sure why), which we also do in the client

BTW, I checked and the `/repo` endpoint does not verify arch and platform, only the Grafana version. The code here sets those headers but as far as I can see those are ignored.